File: tests/benchmarks/resolution/fixtures/lua/repository.lua
local M = {}

local users = {}

function M.save(user)
    users[user.id] = user
end

function M.find_by_id(id)
    return users[id]
end

function M.delete(id)
    if users[id] then
        users[id] = nil
        return true
    end
    return false
end

function M.count()
    local n = 0
    for _ in pairs(users) do
        n = n + 1
    end
    return n
end

return M

File: tests/benchmarks/resolution/fixtures/go/repository.go
package main

import "fmt"

// User represents a domain entity.
type User struct {
	ID    string
	Name  string
	Email string
}

// UserRepository provides persistence for User entities.
type UserRepository struct {
	store map[string]User
}

// NewUserRepository creates and initializes a UserRepository.
func NewUserRepository() *UserRepository {
	return &UserRepository{
		store: make(map[string]User),
	}
}

// FindByID retrieves a user by ID.
func (r *UserRepository) FindByID(id string) (User, bool) {
	u, ok := r.store[id]
	return u, ok
}

// Save persists a user to the store.
func (r *UserRepository) Save(u User) {
	r.store[u.ID] = u
	fmt.Printf("saved user %s\n", u.ID)
}

// Delete removes a user by ID.
func (r *UserRepository) Delete(id string) bool {
	if _, ok := r.store[id]; !ok {
		return false
	}
	delete(r.store, id)
	return true
}

// Count returns the number of stored users.
func (r *UserRepository) Count() int {
	return len(r.store)
}
